### DISCUSSION TITLE
Multiple subnets in left and right

### DISCUSSION THREAD

**Discussion by legolego621** [strongswan/strongswan#657] [Q&A] [answered]
Good afternoon.
Can you please tell me if it is possible to configure several left and right subnets from strongswan for one client?

I am trying to create one peer connection on the client's network equipment to the server, but with two policies

[peer] any_client_ip -> public_ip_address_strongswan_server

10.1.1.1/32 -> 192.168.10.1/32
10.1.1.2/32 -> 192.168.10.2/32

but only the policy that made the connections first always works.
When connecting another policy, errors appear in the strongsvan logs:

```
Oct 08 17:05:53 ipsec1 charon[41759]: 15[ENC] generating CREATE_CHILD_SA response 3 [ N(NO_PROP) ]
Oct 08 17:05:53 ipsec1 charon[41759]: 15[NET] sending packet: from external_ip_server[4500] to external_ip_client[4500] (80 bytes)
Oct 08 17:05:58 ipsec1 charon[41759]: 05[NET] received packet: from external_ip_client[4500] to external_ip_server[4500] (528 bytes)
Oct 08 17:05:58 ipsec1 charon[41759]: 05[ENC] parsed CREATE_CHILD_SA request 4 [ No KE SA TSi TSr ]
Oct 08 17:05:58 ipsec1 charon[41759]: 05[CFG] received proposals: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
Oct 08 17:05:58 ipsec1 charon[41759]: 05[CFG] configured proposals: ESP:AES_CBC_128/AES_CBC_192/AES_CBC_256/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/HMAC_SHA1_96/AES_XCBC_96/NO_EXT_SEQ
Oct 08 17:05:58 ipsec1 charon[41759]: 05[IKE] no acceptable proposal found
Oct 08 17:05:58 ipsec1 charon[41759]: 05[IKE] failed to establish CHILD_SA, keeping IKE_SA
```

Tried specifying the following in right and left, but that didn't solve the problem.

```
       leftsubnets={192.168.10.1/32,192.168.10.2/32} (also with ',' in end)
       rightsubnets=0.0.0.0/0
       
 or
       leftsubnet=0.0.0.0/0
       rightsubnet=0.0.0.0/0
```

       

Strongswan version: 5.8.2
OS: Ubuntu 20.04

Strongswan configuration

```
config setup
        uniqueids=no

conn ikev2
        auto=add
        ikelifetime=12h
        lifetime=12h
        type=tunnel
        keyexchange=ikev2

        left=public_ip_address_strongswan_server
        right=%any

        leftauth=psk
        rightauth=psk
conn a
       also=ikev2

       leftsubnet=192.168.10.1/32,192.168.10.2/32
       rightsubnet=0.0.0.0/0
```

**Comment by tobiasbrunner** [comment, marked as answer]
>  Can you please tell me if it is possible to configure several left and right subnets from strongswan for one client?

Yes, but see [this FAQ entry](https://wiki.strongswan.org/projects/strongswan/wiki/FAQ#Multiple-subnets-per-SA) as some implementations don't support it.

>     Oct 08 17:05:58 ipsec1 charon[41759]: 05[CFG] received proposals: ESP:AES_CBC_256/HMAC_SHA2_256_128/MODP_2048/NO_EXT_SEQ
>     Oct 08 17:05:58 ipsec1 charon[41759]: 05[CFG] configured proposals: ESP:AES_CBC_128/AES_CBC_192/AES_CBC_256/HMAC_SHA2_256_128/HMAC_SHA2_384_192/HMAC_SHA2_512_256/HMAC_SHA1_96/AES_XCBC_96/NO_EXT_SEQ
>     Oct 08 17:05:58 ipsec1 charon[41759]: 05[IKE] no acceptable proposal found

That has nothing to do with your traffic selectors, but, as the error message indicates, the proposals. The peer proposes `modp2048` to do a DH exchange for the CHILD_SA (PFS), but your config does not match that (the default ESP proposal does not include any DH groups). Configure `esp=aes256-sha256-modp2048` to accept the peer's proposal exactly.

**Comment by legolego621** [comment]
Thank you, the problem was exactly in encryption, it is strange that it could not be reproduced with one policy (subnet). One more moment, can I somehow narrow the network on the right side? For example, on the server side, configure left=10.0.0.0/8 and on the client side 10.1.1.1/32?

When used from the server side, left=0.0.0.0/0 everything works, but as described above it does not.

I will be grateful for the answer, but in principle you have already helped me.

**Comment by legolego621** [comment]
I beg your pardon. Everything worked, even when the subnet was narrowed. So the problem was encryption. Thanks for the operative help. Have a nice day.
